### DISCUSSION TITLE
TYPEDSQL - proper way to use UUID?

### DISCUSSION THREAD

**Discussion by RadekHavelka** [prisma/prisma#25247] [Q&A]
### Question

What is the proper way to use UUID in typed SQL new feature in prisma? I have definition of a table

```
model SomeTable {
  id String @id @default(uuid()) @db.Uuid
  title String
  }
```
this is reflected in Postgres as id:uuid, title:text data types.

Then I have typed sql query

```
-- @param {String} $1:id

select id, title
from "SomeTable"
where id = $1
```

trying to run it like this I get an error

Raw query failed. Code: `42883`. Message: `ERROR: operator does not exist: uuid = text
HINT: No operator matches the given name and argument types. You might need to add explicit type casts.`

and I have to go and change it to

```
-- @param {String} $1:id

select id, title
from "SomeTable"
where id = $1::uuid
```

to explicitly cast it to uuid, which I think is not correct, this should be done by the very typedsql. What am I doing wrong then?

PS. this is of course massivly simplified example, one would never do this kind of query via rawQuery ...

### Information about Prisma Schema, Client Queries and Environment (optional)

Prisma 5.19.1 + Posgres 14 + Node 22

**Comment by ludralph** [comment]
Hi @RadekHavelka 👋 

Thank you for raising this question. I'm checking with the team internally to ascertain the correct way to use `uuid` within typedSql. I'll provide a response as soon as possible.

**Reply by ludralph** [reply]
Hi @RadekHavelka 

In an ideal scenario, TypedSQL should work out of the box with UUID without requiring explicit type casting. However, PostgreSQL often infers UUIDs as strings unless they are explicitly cast to the uuid type.

**Reply by lookto** [reply]
Hi @ludralph,

is there any progress regarding this issue?

**Reply by Spoutnik97** [reply]
@lookto you can use it like this in your .sql file : 

```
// userQuery.sql

user_id = $1::uuid

// getUser.ts

prisma.$queryRawTyped(userQuery(id))
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
